src/github.com/getlantern/detour/detour.go

```diff
@@ -369,60 +365,15 @@ var nonIdempotentMethods = [][]byte{
 	[]byte("PATCH "),
 }
 
-// ref section 9.1.2 of https://www.ietf.org/rfc/rfc2616.txt.
-// checks against non-idemponent methods actually,
-// as we consider the https handshake phase to be idemponent.
-func (dc *Conn) isIdempotentRequest() bool {
-	dc.muLocalBuffer.Lock()
-	defer dc.muLocalBuffer.Unlock()
-	b := dc.localBuffer.Bytes()
+// Ref section 9.1.2 of https://www.ietf.org/rfc/rfc2616.txt.
+// We consider the https handshake phase to be idemponent.
+func isNonIdempotentHTTPRequest(b []byte) bool {
```

Comment:
`GET /posts/delete?id=123` is still idempotent as long as `ID` is an actual ID representing an unique resource instead of a sequence number. Something like `GET /posts/create` is the problem.

Since we can't enumerate all applications behaviour, the least thing we can do is to follow standard. Just like browser not prompting user when refreshing a GET request, we can do the same I think.